todoapp/app/src/main/java/com/example/android/architecture/blueprints/todoapp/tasks/TasksFragment.kt

```diff
@@ -89,27 +87,31 @@ class TasksFragment : Fragment() {
     }
 
     private fun showFilteringPopUpMenu() {
-        PopupMenu(context, activity.findViewById<View>(R.id.menu_filter)).run {
-            menuInflater.inflate(R.menu.filter_tasks, menu)
-
-            setOnMenuItemClickListener {
-                viewDataBinding.viewmodel?.run {
-                    currentFiltering =
-                            when (it.itemId) {
-                                R.id.active -> TasksFilterType.ACTIVE_TASKS
-                                R.id.completed -> TasksFilterType.COMPLETED_TASKS
-                                else -> TasksFilterType.ALL_TASKS
-                            }
-                    loadTasks(false)
+        context?.let { cxt ->
+            activity?.let { act ->
+                PopupMenu(cxt, act.findViewById<View>(R.id.menu_filter)).run {
+                    menuInflater.inflate(R.menu.filter_tasks, menu)
+
+                    setOnMenuItemClickListener {
+                        viewDataBinding.viewmodel?.run {
+                            currentFiltering =
+                                    when (it.itemId) {
+                                        R.id.active -> TasksFilterType.ACTIVE_TASKS
+                                        R.id.completed -> TasksFilterType.COMPLETED_TASKS
+                                        else -> TasksFilterType.ALL_TASKS
+                                    }
+                            loadTasks(false)
+                        }
+                        true
+                    }
+                    show()
                 }
-                true
             }
-            show()
         }
     }
 
     private fun setupFab() {
-        activity.findViewById<FloatingActionButton>(R.id.fab_add_task).run {
+        activity?.findViewById<FloatingActionButton>(R.id.fab_add_task)?.apply {
```

Comment:
Revert back to `run` (try to minimize changes and focus on modifying for nullablility in 27)

Reply:
You're right, the ```run``` is here better than ```apply``` .